controller/project/manager.py

```diff
@@ -219,12 +229,186 @@ def __get_crowd_label_is_data(project_id: str, is_id: str) -> Dict[str, str]:
+    for is_item in is_items:
+        if is_item.type != enums.InformationSourceType.ACTIVE_LEARNING.value:
+            # only active learning information sources are pickled
+            continue
+        is_id = str(is_item.id)
+        last_payload = information_source.get_last_payload(project_id, is_id)
+        if last_payload.state == enums.PayloadState.FINISHED.value:
+            al_path = os.path.join(
+                "/inference", project_id, f"active-learner-{is_id}.pkl"
+            )
+            if not os.path.exists(al_path):
+                missing.append(is_id)
+    return missing
+
+
+def update_project_for_gates(project_id: str, user_id: str) -> None:
+
+    project_item = project.get(project_id)
+    if not project_item:
+        return
+
+    global projects_updating
+    with projects_updating_lock:
+        if project_id in projects_updating:
+            return
+        projects_updating.add(project_id)
+
+    notification.send_organization_update(project_id, "gates_integration:started")
+
+    if not __tokenizer_pickle_exists(project_item.tokenizer):
+        daemon.run(request_save_tokenizer, project_item.tokenizer)
+
+    missing_emb_pickles = __get_missing_embedding_pickles(project_id)
+    for embedding_id in missing_emb_pickles:
+        __create_embedding_pickle(project_id, embedding_id, user_id)
+
+    # removes the project from the set of projects that are currently updating
+    # when the function returns
+    daemon.run(
+        __wait_and_create_information_source_pickles,
+        project_id,
+        user_id,
+    )
+
+
+def __create_embedding_pickle(project_id: str, embedding_id: str, user_id: str):
+
+    embedding_item = embedding.get(project_id, embedding_id)
+    if not embedding_item:
+        return
+
+    request_deleting_embedding(project_id, embedding_id)
+
+    attribute_id = str(embedding_item.attribute_id)
+    attribute_name = attribute.get(project_id, attribute_id).name
+    if embedding_item.type == enums.EmbeddingType.ON_ATTRIBUTE.value:
+        prefix = f"{attribute_name}-classification-"
+        config_string = embedding_item.name[len(prefix) :]
+        request_creating_attribute_level_embedding(
+            project_id, attribute_id, user_id, config_string
+        )
+    else:
+        prefix = f"{attribute_name}-extraction-"
+        config_string = embedding_item.name[len(prefix) :]
+        request_creating_token_level_embedding(
+            project_id, attribute_id, user_id, config_string
+        )
+
+
+def __wait_and_create_information_source_pickles(
+    project_id: str,
+    user_id: str,
+) -> None:
+    # wait to make sure db entries for embeddings are created
+    time.sleep(5)
+    __wait_for_tokenizer(project_id)
+    __wait_for_embeddings(project_id)
+
+    missing_is_ids = __get_missing_information_source_pickles(project_id)
+    for is_id in missing_is_ids:
+        payload_manager.create_payload(project_id, is_id, user_id)
```

Comment:
same as with embeddings, might overload if we have multiple big active learners in the project so maybe max 2 at a time?

Reply:
I think it is not as problematic as the embedders. Running all heuristics at the same time is also possible via the UI, so we should also have an eye on that. I added it to the findings.

Reply:
yeah there already is an issue for some kind of queue logic for the run all because it often runs into issues :)